src/main/java/utilities/interceptors/RetryInterceptor.java

```diff
@@ -37,12 +37,16 @@ public Response intercept(Chain chain) throws IOException {
 		return response;
 	}
 	
-	private Response doRequest(Chain chain, Request request){
+	private Response doRequest(Chain chain, Request request) throws IOException {
```

Comment:
Shouldn't this list out *ALL* exception types that are being thrown?

I know that NPE is a child of IOE but you also have another catch for other exception types. Won't that cause this method to throw other types also?